### PR TITLE
Improved: Progress bar color logic for Transfer Orders when scanning items

### LINKED ISSUE #461: Show overscanning alert text in toast.
This is the feature request:
```When scanning items and you exceed the qty ordered, the line turns red. However, when manually entering a qty that exceeds the ordered qty, a message appears indicating that “the picked qty cannot exceed the ordered qty. X already shipped”. > Can we add this messaging when over scanning and not only when manually entering the qty?```

I think if a user is scanning an over scans an item then we should show a toast that over fulfilling is not allowed. However I am not sure if the helper text can be shown if we do not add the qty to the input field. If the qty is added to the input field then it should show the alert text as well.

### PR DESCRIPTION
**Pull request by Chinmay2107**
Changelog: Changed

### Related Issues
Closes #461 

### Short Description and Why It's Useful
When scanning items if the scanned item quantity exceeds the qty ordered, the line turns red and a toast appears indicating that “the picked qty cannot exceed the ordered qty. X already shipped”. A toast will appear when this scenario occurs.
I've also changed the color of the Progress Bar:
1) Yellow when an item is partially scanned.
2) Green when the requested qty is scanned.
3) Red when quantity exceeds the requested quantity.

### Screenshots of Visual Changes before/after (If There Are Any)
Screenshots supporting the modifications made in the progress bar's color:
1) Yellow progress bar in case of partial completion of scanning
Screenshot from 2024-05-23 17-34-22
Screenshot from 2024-05-23 17-34-36
2) Green progress bar in case of partial completion of scanning
Screenshot from 2024-05-23 17-34-46
3) Red progress bar in case of partial completion of scanning
Screenshot from 2024-05-23 17-34-52

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "fulfillment",
-  "version": "2.4.2",
+  "version": "2.4.3",
```
Comment (ymaheshwari1) [inline package.json:3]: We do not update the app version for each change, revert this change

File: src/components/TransferOrderItem.vue [hunk 2]
```diff
@@ -31,7 +31,7 @@
       </div>
 
       <div class="qty-progress">
-        <ion-progress-bar :color="getPickedToOrderedFraction(item) > 1 ? 'danger' : 'primary'" :value="getPickedToOrderedFraction(item)" />
+        <ion-progress-bar :color="setProgressBarColor(item)" :value="getPickedToOrderedFraction(item)" />
```
Comment (ymaheshwari1) [inline src/components/TransferOrderItem.vue:34]: We are getting the progressBar color on the basis of some condition, so it can be renamed to something like `getProgessBarColor`

File: src/components/TransferOrderItem.vue [hunk 3]
```diff
@@ -100,6 +100,12 @@ export default defineComponent({
     }),
   },
   methods: {
+    setProgressBarColor(event:any, item: any) {
+         const fraction = this.getPickedToOrderedFraction(item);
```
Comment (ymaheshwari1) [inline src/components/TransferOrderItem.vue:104]: Improve indentation, we follow indentation of two spaces across app.

File: src/components/TransferOrderItem.vue [hunk 4]
```diff
@@ -100,6 +100,12 @@ export default defineComponent({
     }),
   },
   methods: {
+    setProgressBarColor(event:any, item: any) {
```
Comment (ymaheshwari1) [inline src/components/TransferOrderItem.vue:103]: No need to define event as its not used anywhere in the method.

File: src/views/TransferOrderDetail.vue [hunk 5]
```diff
@@ -295,6 +295,13 @@
           showToast(translate("Scanned item is already completed:", { itemName: payload }))
         }
         else if (result.isProductFound) {
+          const item = result.orderItem
+          const shippedQuantity = this.getShippedQuantity(item)
+          if(item.pickedQuantity > item.orderedQuantity - shippedQuantity)
+          {
```
Comment (ymaheshwari1) [inline src/views/TransferOrderDetail.vue:301]: Improve indentation

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

**Comment by ymaheshwari1** [comment]
@Chinmay2107 Is this done from your end? As I remember we are waiting here to conclude that how the error needs to be displayed when user is entering the quantity manually in the input field.

**Comment by Chinmay2107** [comment]
> @Chinmay2107 Is this done from your end? As I remember we are waiting here to conclude that how the error needs to be displayed when user is entering the quantity manually in the input field.

Yes, the changes are done from my end.

File: package.json (hunk 1 above)
Comment (Chinmay2107) [reply package.json]: I have changed the version.

File: src/components/TransferOrderItem.vue (hunk 2 above)
Comment (Chinmay2107) [reply src/components/TransferOrderItem.vue]: Updated the method name to getProgressBarColor.

File: src/components/TransferOrderItem.vue (hunk 3 above)
Comment (Chinmay2107) [reply src/components/TransferOrderItem.vue]: I have improved the indentation.

File: src/components/TransferOrderItem.vue (hunk 4 above)
Comment (Chinmay2107) [reply src/components/TransferOrderItem.vue]: Done

File: src/views/TransferOrderDetail.vue (hunk 5 above)
Comment (Chinmay2107) [reply src/views/TransferOrderDetail.vue]: Corrected indentation part.

**Review by ymaheshwari1 — APPROVED** [review]
